specs/ceph_installer_wrapper.adoc

```diff
@@ -37,103 +37,330 @@ A wrapper needs to be created which abstracts the capabilities provided by the c
+      "journal_colocation": "false",
+      "storage_disks": [
+        {
+          "device": "/dev/sda",
+          "journal": "/dev/sdc"
+        },
+        {
+          "dev_name": "/dev/sdb",
+          "journal": "/dev/sdc"
+        }
+      ]
+    },
+    "10.0.0.30": {
+      "role": "ceph/osd",
+      "provisioning_ip": "10.0.0.30",
+      "journal_colocation": "true",
+      "storage_disks": [
+        {
+          "device": "/dev/sda"
+        },
+        {
+          "dev_name": "/dev/sdb"
+        }
+      ]
+    }
+  }
+}
+----
 
-One of the ceph monitors will be marked as a provisioner node for the cluster and the ceph-installer to be installed on the node. This node is expected to accept all ceph provisioning related tasks.
 
-=== Developer impact:
+=== Tendrl/node_agent impact:
 
-None.
+Since ceph installer does not support operating from a node inside the cluster,
+the node agent running on the API node needs to be designated as a provisioner
+for ceph cluster. This would be accomplished by adding a tag for
+`provisioners/ceph`. The API would route jobs directly to this tag, with an
+empty `node_ids` array in the job payload.
 
+[source,json]
+.Job structure with all the possible parameters
+----
+{
+  "integration_id": "9a4b84e0-17b3-4543-af9f-e42000c52bfc",
+  "run": "tendrl.flows.CreateCluster",
+  "status": "new",
+  "type": "node",
+  "node_ids": [],
+  "tags": ["provisioner/ceph"],
+  "parameters": {
+    "sds_type": "ceph",
+    "sds_version": "10.2.5",
+    "name": "MyCluster",
+    "TendrlContext.integration_id": "9a4b84e0-17b3-4543-af9f-e42000c52bfc",
+    "Node[]": [
+      "3a95fd96-876d-439a-a64d-70332c069aaa",
+      "3943fab1-9ed2-4eb6-8121-5a69499c4568",
+      "b10e00e9-e444-41c2-9517-df2118b42731"
+    ],
+    "node_configuration": {
+      "3a95fd96-876d-439a-a64d-70332c069aaa": {
+        "role": "ceph/mon",
+        "provisioning_ip": "10.0.0.24"
+      },
+      "3943fab1-9ed2-4eb6-8121-5a69499c4568": {
+        "role": "ceph/osd",
+        "provisioning_ip": "10.0.0.29",
+        "journal_size": 5192,  # Megabytes
+        "journal_colocation": "false",
+        "storage_disks": [
+          {
+            "device": "/dev/sda",
+            "journal": "/dev/sdc"
+          },
+          {
+            "dev_name": "/dev/sdb",
+            "journal": "/dev/sdc"
+          }
+        ]
+      },
+      "b10e00e9-e444-41c2-9517-df2118b42731": {
+        "role": "ceph/osd",
+        "provisioning_ip": "10.0.0.30",
+        "journal_colocation": "true",
+        "storage_disks": [
+          {
+            "device": "/dev/sda"
+          },
+          {
+            "dev_name": "/dev/sdb"
+          }
+        ]
+      }
+    }
+  }
+}
+----
+
+The validations to be carried out on these parameters are as follows, when the
+role of a node is `ceph/osd`:
+
+* `journal_size` is optional. Value should be in Megabytes.
+* `journal_colocation` is optional and value should be a boolean.
```

Comment:
Is `"journal_colocation": "true"` and `"journal": "/dev/sdc"` valid configuration?